llvm/tools/gold/gold-plugin.cpp

```diff
@@ -308,6 +314,14 @@ namespace options {
       RemarksFormat = std::string(opt);
     } else if (opt.consume_front("stats-file=")) {
       stats_file = std::string(opt);
+    } else if (opt.consume_front("time-trace=")) {
+      time_trace_file = std::string(opt);
+    } else if (opt.consume_front("time-trace-granularity=")) {
+      unsigned Granularity;
+      if (opt.getAsInteger(10, Granularity))
+        message(LDPL_FATAL, "Invalid time trace granularity: %s", opt);
```

Comment:
This is passing a StringRef to a varargs function looking for `const char *`. This is obviously wrong, but it looks like the ABI happens to work out fine on non-s390x architectures...

Reply:
I've put up https://github.com/llvm/llvm-project/pull/95083 to fix this.